lib/spack/spack/test/builder.py

```diff
@@ -123,6 +124,9 @@ def test_old_style_compatibility_with_super(spec_str, method_name, expected):
     assert value == expected
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32", reason="log_output to file conflicts with pytest on Windows"
+)
```

Comment:
Should this have a TODO comment to address (I'm fine skipping for now)

Also I want to make sure this is a log_output issue, since the checks appear to be on setting env variables and not dependent on reading output.

Reply:
It does not depend on the log_output, but non the less that method is invoked over the course of this test as a context manager, which causes the stdout failure we discussed.

Reply:
Oh I see, we never actually get to the assert. In general this must be working (e.g. for install tests that perform this in a subprocess) so I guess there's an issue with using `log_output` in the current process on Windows; if that sounds right to you, could you refine the `reason` accordingly?

Reply:
Actually Markus uncovered an instance of this in the installer tests yesterday, so it does seem like it hits there too (albeit inconsistently). But I'll have the reason updated.